frontends/election-manager/src/components/election_manager_writein_tally_report.tsx

```diff
@@ -76,8 +89,13 @@ export const ElectionManagerWriteInTallyReport = forwardRef<
 
     return (
       <TallyReport className="print-only" ref={ref}>
-        {ballotStylePartyIds.map((partyId) =>
-          precinctIds.map((precinctId) => {
+        {ballotStylePartyIds.map((partyId) => {
+          const writeInCountsFilteredByParty = filterWriteInCountsByParty(
+            writeInCounts,
+            election,
+            partyId
+          );
```

Comment:
There was previously a bug where all contests were appearing on both party pages. In other words, we were not filtering contests by party.